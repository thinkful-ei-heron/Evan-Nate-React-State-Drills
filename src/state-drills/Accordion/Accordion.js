import React, {Component} from 'react';

export default class Accordion extends Component {
    static defaultProps = {
        sections: []
    }

    state = {
       currentSectionIndex: null 
    }

    changeSectionIndex = (Index) => {
        if(Index === this.state.currentSectionIndex){
            this.setState({currentSectionIndex: null}); 
        }
        else{
        this.setState ({ currentSectionIndex: Index})
        }
    }

    renderContent(section, index, currentSectionIndex) {
        return (
            <li key={index}>
                <button type='button' onClick={() => this.changeSectionIndex(index)}>{section.title}</button>
                {(currentSectionIndex === index) && <p>{section.content}</p>}
            </li>
        )
    }

    render() {
        const {currentSectionIndex} = this.state
        const {sections} = this.props
        return (
            <ul>
                {sections.map((section, index) =>
                    this.renderContent(section, index, currentSectionIndex))}
            </ul>
        )
    }
}
