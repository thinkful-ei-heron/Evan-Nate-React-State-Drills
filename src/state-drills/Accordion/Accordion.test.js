import React from 'react';
import ReactDom from 'react-dom';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe(`Accordion Component`, () => {
  const sectionsProp = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]
  it('renders without errors', () => {
    const div = document.createElement('div'); 
    ReactDom.render (<Accordion/>, div);
    ReactDom.unmountComponentAtNode(div)
  })

  it(`renders empty when given no sections`, () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders none of the sections by default', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('closes the first section and opens any clicked sections', () => {
    const wrapper = shallow(<Accordion sections={sectionsProp} /> )
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

