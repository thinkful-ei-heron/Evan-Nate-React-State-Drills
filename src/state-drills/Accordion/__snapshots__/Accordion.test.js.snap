// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component closes the first section and opens any clicked sections 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders empty when given no sections 1`] = `<ul />`;

exports[`Accordion Component renders none of the sections by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Section 3
    </button>
  </li>
</ul>
`;
